src/mame/misc/gameace.cpp

```diff
@@ -56,103 +66,336 @@ class gameace_state : public driver_device
+{
+	uint8_t* rom = memregion("maincpu")->base();
+	m_membank->configure_entries(0, 0x10, &rom[0x00000], 0x4000);
+	m_membank->set_entry(7);
+	m_videoview.select(0);
+	m_palview.select(0);
+}
+
+void gameace_state::draw_sprites(bitmap_ind16& bitmap, const rectangle& cliprect)
+{
+	// very similar to mitchell.cpp, but with everything on a different byte
+	for (int offs = 0x1000 - 0x40; offs >= 0; offs -= 0x20)
+	{
+		int code = m_spriteram[offs + 0x10];
+		int sy = 255 - m_spriteram[offs + 0x11];
+		int attr = m_spriteram[offs + 0x13];
+		int color = attr & 0x0f;
+		int sx = m_spriteram[offs + 0x12] + ((attr & 0x10) << 4);
+		code += (attr & 0xe0) << 3;
+		/*
+		if (m_flipscreen)
+		{
+			sx = 496 - sx;
+			sy = 240 - sy;
+		}
+		*/
+		m_gfxdecode->gfx(1)->transpen(bitmap, cliprect,
+			code,
+			color,
+			0, 0,
+			sx-7, sy-14, 15);
+	}
+}
+
 
 uint32_t gameace_state::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
 {
+	m_fg_tilemap->draw(screen, bitmap, cliprect, 0, 0);
+	draw_sprites(bitmap, cliprect);
 	return 0;
 }
 
+TILE_GET_INFO_MEMBER(gameace_state::get_fg_tile_info)
+{
+	int code = m_fgram[tile_index*2] | (m_fgram[(tile_index*2)+1] << 8);
+	int col = m_colram[tile_index];
+	tileinfo.set(0,code,col,0);
+}
+
+void gameace_state::fgram_w(offs_t offset, uint8_t data)
+{
+	m_fgram[offset] = data;
+	m_fg_tilemap->mark_tile_dirty(offset/2);
+}
+
+void gameace_state::colram_w(offs_t offset, uint8_t data)
+{
+	m_colram[offset] = data;
+	m_fg_tilemap->mark_tile_dirty(offset);
+}
+
+void gameace_state::video_start()
+{
+	m_fg_tilemap = &machine().tilemap().create(*m_gfxdecode, tilemap_get_info_delegate(*this, FUNC(gameace_state::get_fg_tile_info)), TILEMAP_SCAN_ROWS, 8, 8, 64, 32);
+
+	m_paletteram.resize(2 * m_palette->entries());
+	m_palette->basemem().set(m_paletteram, ENDIANNESS_LITTLE, 2);
+}
+
+
+void gameace_state::bank_w(uint8_t data)
+{
+	if (data & 0xf0)
+		logerror("bank_w unused bits %02x\n", data & 0xf0);
+
+	m_membank->set_entry(data & 0xf);
+}
+
+void gameace_state::palbank_w(uint8_t data)
+{
+	if (data & 0xdf)
+		logerror("palbank_w unused bits %02x\n", data & 0xdf);
+
+	m_palview.select((data & 0x20)>>5);
+}
+
+
+void gameace_state::vidbank_w(uint8_t data)
+{
+	if (data & 0xfe)
+		logerror("videbank_w unused bits %02x\n", data & 0xfe);
+
+	m_videoview.select(data & 1);
+}
+
+uint8_t gameace_state::pal_low_r(offs_t offset, uint8_t data)
+{
+	return m_palette->read8(offset);
+}
+
+uint8_t gameace_state::pal_high_r(offs_t offset, uint8_t data)
+{
+	return m_palette->read8(offset+0x800);
+}
+
+void gameace_state::pal_low_w(offs_t offset, uint8_t data)
+{
+	m_palette->write8(offset, data);
+}
+
+void gameace_state::pal_high_w(offs_t offset, uint8_t data)
+{
+	m_palette->write8(offset+0x800, data);
+}
 
 void gameace_state::main_program_map(address_map &map)
 {
-	map(0x0000, 0x8000).rom().region("maincpu", 0);
+	map(0x0000, 0x7fff).rom().region("maincpu", 0);
+	map(0x8000, 0xbfff).bankr("bank1");
```

Comment:
You can put `m_membank` here directly to avoid the potential need to update the literal tag in multiple places.